### DISCUSSION TITLE
Can a RandomizableTransform depend on a previous Transform?

### DISCUSSION THREAD

**Discussion by jonieva** [Project-MONAI/MONAI#7062] [Q&A] [answered]
Hi there!

I was wondering if there is a way to make a RandomizableTransform depend on a previous RandomizableTransform during a training pipeline.

During training, I would like to do something like:
- Apply RandomizableTransformA
- Apply RandomizableTransformB
- If RandomizableTransformB was not applied to this image, apply RandomizableTransformC.

I know I can override the randomize method in my transform, but I'm not sure how could I know if a previous transform in the Compose pipeline was applied.

A workaround could be to implement a "composed Transform" RandomizableTransform_B_or_C, but I was wondering if someone has a better alternative.

**Comment by wyli** [comment, marked as answer]
thanks for the discussion, does the `OneOf` transform as an extension of `Compose` solve this problem? https://github.com/Project-MONAI/MONAI/blob/f140e062355c8565df180e31ad9f91f64299c4de/monai/transforms/compose.py#L385-L388

**Comment by jonieva** [comment]
I think it does! Or at least it's definitely much cleaner :-)

Thank you very much @wyli
